dali/pipeline/data/tensor_vector.h

```diff
@@ -148,6 +148,26 @@ class TensorVector {
     return type_;
   }
 
+  inline void SetLayout(DALITensorLayout layout) {
+    if (state_ == State::noncontiguous) {
+      DALI_ENFORCE(!tensors_.empty(), "Layout cannot be set uniformly for empty batch");
+    }
+    tl_->SetLayout(layout);
+    for (auto t : tensors_) {
+      t->SetLayout(layout);
+    }
+  }
+
+  inline DALITensorLayout GetLayout() const {
+    if (state_ == State::contiguous) {
+      return tl_->GetLayout();
+    }
+    if (tensors_.size() > 0) {
+      return tensors_[0]->GetLayout();
+    }
+    return DALITensorLayout::DALI_UNKNOWN;
```

Comment:
I'd prefer `return tensors_.size() > 0 ? tensors_[0]->GetLayout() : DALITensorLayout::DALI_UNKNOWN;`

Reply:
It isn't that clear why only one of the ifs would be turned into ternary, and also it would require a line break and probably look ugly.

Reply:
I agree with @klecki  - now we have 3 clear cases. Ternary would merge two into one (at least visually).